src/Config/Validator/Deploy/PhpVersion/ConstraintFactory.php

```diff
@@ -0,0 +1,49 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+namespace Magento\MagentoCloud\Config\Validator\Deploy\PhpVersion;
+
+use Composer\Semver\Constraint\Constraint;
+use Composer\Semver\Constraint\ConstraintInterface;
+use Composer\Semver\Constraint\MultiConstraint;
+use Magento\MagentoCloud\App\ContainerInterface;
+
+/**
+ * Creates instance of ConstraintInterface object
+ */
+class ConstraintFactory
+{
+    /**
+     * @var ContainerInterface
+     */
+    private $container;
+
+    /**
+     * @param ContainerInterface $container
+     */
+    public function __construct(ContainerInterface $container)
+    {
+        $this->container = $container;
+    }
+
+    /**
+     * @param string $operator
+     * @param string $version
+     * @return Constraint
+     */
+    public function constraint(string $operator, string $version): Constraint
+    {
+        return $this->container->create(Constraint::class, [$operator, $version]);
```

Comment:
This causing an exception:
`Unresolvable dependency resolving [Parameter #0 [ <required> $operator ]] in class Composer\Semver\Constraint\Constraint`
Tried locally and on cloud environment.

Reply:
Done